[Docs] Expanding details on the upgrade process (#126)
cloudbees/terraform-aws-cloudbees-ci-eks-addon: CONTRIBUTING.md

## Release

CloudBees CI Terraform EKS Addon versions try to be in sync with the [CloudBees CI releases](https://docs.cloudbees.com/docs/release-notes/latest/cloudbees-ci/).

Before a new CloudBees CI Helm chart is released, new features for this addon and its companion blueprints are merged into the integration branch (`develop`). When a new version of the CloudBees CI Helm chart is released, the addon is updated to the new version following this process:

1. Test the update in the integration branch (`develop`)
   - Update The field `version` in CloudBees CI Terraform EKS Addon update needs to be updated to the new [version of the Helm chart](https://artifacthub.io/packages/helm/cloudbees/cloudbees-core/)

>    - The field `source` in the `eks_blueprints_addon_cbci` at blueprints folder must point to the local to the CloudBees CI Terraform EKS Addon the root of the repository `source = "../../"` (not to the remote [terraform registry version](https://registry.terraform.io/modules/cloudbees/cloudbees-ci-eks-addon/aws/latest))

Reviewer: ```suggestion
   - The field `source` in the `eks_blueprints_addon_cbci` in the blueprints folder must point to the local CloudBees CI Terraform EKS Addon root of the repository `source = "../../"` (not to the remote [terraform registry version](https://registry.terraform.io/modules/cloudbees/cloudbees-ci-eks-addon/aws/latest)).
```
Please check to make sure I did not change the meaning.

>    - Test the update locally

Reviewer: ```suggestion
   - Test the update locally.
```

> [!TIP]
> Use the following targets from [Makefile](Makefile): `deploy` > `validate` > `destroy`.

> 2. Create a PR against the `main` branch including the Helm chart update plus other updates available in the integration branch (`develop`) ensuring the field `source` in the `eks_blueprints_addon_cbci` at blueprints is pointing to the remote [terraform registry version](https://registry.terraform.io/modules/cloudbees/cloudbees-ci-eks-addon/aws/latest) and `version >= "x.x.x"`.

Reviewer: ```suggestion
2. Create a PR against the `main` branch. including the Helm chart update plus other updates available in the integration branch (`develop`). Ensure that the field `source` in the `eks_blueprints_addon_cbci` at blueprints is pointing to the remote [terraform registry version](https://registry.terraform.io/modules/cloudbees/cloudbees-ci-eks-addon/aws/latest) and `version >= "x.x.x"`.
```
When you say "at blueprints" I'm not sure what you mean. Do you mean in the blueprints folder? If your user already knows this, ignore my comment. :)

3. Once the pull request is merged, verify that the `main` branch successfully passes the [Terraform CI build](#blueprint-terraform-ci-pipeline).

> 4. Create a [new release](https://github.com/cloudbees/terraform-aws-cloudbees-ci-eks-addon/releases). The release version semantics follows the Helm chart convention.

Reviewer: ```suggestion
4. Create a [new release](https://github.com/cloudbees/terraform-aws-cloudbees-ci-eks-addon/releases). The release version semantics follow the Helm chart convention.
```
